File: source-code/ui.py
import os
import gi
gi.require_version('Gtk', '4.0')
gi.require_version('Adw', '1')
from gi.repository import Gtk, Adw, GdkPixbuf

def build_ui(window):
    # HeaderBar dla profesjonalnego wyglądu
    header_bar = Adw.HeaderBar()
    header_bar.set_show_end_title_buttons(True)

    # Box dla title widget z logo i tytułem
    title_box = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL, spacing=10)

    # Logo (mniejsze dla headera)
    logo_path = "/usr/share/HackerOS/ICONS/Plymouth-Icons/watermark.png"
    if os.path.exists(logo_path):
        pixbuf = GdkPixbuf.Pixbuf.new_from_file_at_size(logo_path, 32, 32)
        logo_image = Gtk.Image.new_from_pixbuf(pixbuf)
    else:
        logo_image = Gtk.Image()  # Puste jeśli nie znaleziono
    title_box.append(logo_image)

    # Tytuł (mniejszy font dla headera)
    title_label = Gtk.Label(label="Witaj w HackerOS!")
    title_label.set_markup("<span font='Arial bold 20'>Witaj w HackerOS!</span>")
    title_box.append(title_label)

    header_bar.set_title_widget(title_box)

    # Content box
    content_box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=15)
    content_box.set_margin_top(20)
    content_box.set_margin_start(20)
    content_box.set_margin_end(20)
    content_box.set_margin_bottom(20)

    # Podtytuł (używany też do feedbacku)
    window.subtitle_label = Gtk.Label(label="Twój system do Gier i Etycznego Hakowania")
    window.subtitle_label.set_markup("<span font='Arial 18'>Twój system do Gier i Etycznego Hakowania</span>")
    window.subtitle_label.set_halign(Gtk.Align.CENTER)
    window.subtitle_label.get_style_context().add_class("subtitle")
    content_box.append(window.subtitle_label)

    # Separator
    separator = Gtk.Separator(orientation=Gtk.Orientation.HORIZONTAL)
    separator.set_margin_start(10)
    separator.set_margin_end(10)
    content_box.append(separator)

    # ScrolledWindow dla przycisków, aby było scrollowalne jeśli potrzeba
    scrolled_window = Gtk.ScrolledWindow()
    scrolled_window.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
    scrolled_window.set_hexpand(True)
    scrolled_window.set_vexpand(True)
    content_box.append(scrolled_window)

    # Layout przycisków - FlowBox dla dynamicznego układu
    flow_box = Gtk.FlowBox()
    flow_box.set_column_spacing(15)
    flow_box.set_row_spacing(15)
    flow_box.set_margin_start(10)
    flow_box.set_margin_end(10)
    flow_box.set_margin_bottom(20)
    flow_box.set_homogeneous(True)
    flow_box.set_min_children_per_line(2)
    flow_box.set_max_children_per_line(3)
    scrolled_window.set_child(flow_box)

    # Pozostałe przyciski (usunięto wskazane)
    buttons = [
        ("Otwórz stronę HackerOS", window.actions.open_website),
        ("Otwórz X", window.actions.open_x),
        ("Otwórz sklep z aplikacjami", window.actions.open_software),
        ("Changelog", window.actions.open_changelog),
        ("Informacje o systemie", window.actions.open_system_info),
        ("Zgłoś błąd", window.actions.report_bug),
        ("Forum dyskusyjne", window.actions.open_forum),
        ("Zaktualizuj system", window.actions.update_system),
        ("Uruchom HackerOS Games", lambda: window.actions.run_command_with_feedback("/usr/share/HackerOS/Scripts/Bin/HackerOS-Games.sh")),
        ("Hacker Launcher", lambda: window.actions.run_command_with_feedback("/usr/share/HackerOS/Scripts/HackerOS-Apps/Hacker_Launcher"))
    ]
    for text, action in buttons:
        btn = Gtk.Button(label=text)
        btn.connect("clicked", lambda widget, act=action: act())
        flow_box.append(btn)

    # Footer
    footer_label = Gtk.Label(label="© 2025 HackerOS Team | All rights reserved")
    footer_label.set_halign(Gtk.Align.CENTER)
    footer_label.get_style_context().add_class("footer")
    content_box.append(footer_label)

    # ToolbarView do integracji headera i contentu
    toolbar_view = Adw.ToolbarView()
    toolbar_view.add_top_bar(header_bar)
    toolbar_view.set_content(content_box)

    # Ustaw content okna
    window.set_content(toolbar_view)

File: source-code/src/ui.rs
use adw::prelude::*;
use adw::ApplicationWindow;
use gtk::{Box as GtkBox, Button, Label, Orientation, ScrolledWindow, Separator};
use std::rc::Rc;
use std::cell::RefCell;

use crate::actions::Actions;

pub fn build_ui(window: &ApplicationWindow, actions_rc: Rc<RefCell<Actions>>) {
    let main_box = GtkBox::builder()
    .orientation(Orientation::Vertical)
    .spacing(20)
    .margin_start(20)
    .margin_end(20)
    .margin_top(20)
    .margin_bottom(20)
    .build();

    // Title
    let title_label = Label::builder()
    .label("Witaj w HackerOS!")
    .css_classes(vec!["title".to_string()])
    .build();
    main_box.append(&title_label);

    // Subtitle (feedback label)
    let subtitle_label = Label::builder()
    .label("Wybierz akcję poniżej.")
    .css_classes(vec!["subtitle".to_string()])
    .build();
    main_box.append(&subtitle_label);

    // Set subtitle_label in Actions
    {
        let mut actions = actions_rc.borrow_mut();
        actions.subtitle_label = Some(subtitle_label);
    }

    // Separator
    let separator1 = Separator::new(Orientation::Horizontal);
    main_box.append(&separator1);

    // Scrolled content for buttons
    let scrolled_window = ScrolledWindow::builder()
    .vexpand(true)
    .build();

    let buttons_box = GtkBox::builder()
    .orientation(Orientation::Vertical)
    .spacing(10)
    .margin_start(10)
    .margin_end(10)
    .margin_top(10)
    .margin_bottom(10)
    .build();

    // Buttons
    let website_button = Button::builder()
    .label("Otwórz stronę HackerOS")
    .build();
    website_button.connect_clicked({
        let actions_rc = actions_rc.clone();
        move |_| actions_rc.borrow().open_website()
    });
    buttons_box.append(&website_button);

    let x_button = Button::builder()
    .label("Otwórz X (Twitter)")
    .build();
    x_button.connect_clicked({
        let actions_rc = actions_rc.clone();
        move |_| actions_rc.borrow().open_x()
    });
    buttons_box.append(&x_button);

    let software_button = Button::builder()
    .label("Otwórz Sklep z aplikacjami")
    .build();
    software_button.connect_clicked({
        let actions_rc = actions_rc.clone();
        move |_| actions_rc.borrow().open_software()
    });
    buttons_box.append(&software_button);

    let changelog_button = Button::builder()
    .label("Otwórz Changelog")
    .build();
    changelog_button.connect_clicked({
        let actions_rc = actions_rc.clone();
        move |_| actions_rc.borrow().open_changelog()
    });
    buttons_box.append(&changelog_button);

    let system_info_button = Button::builder()
    .label("Otwórz Informacje o systemie")
    .build();
    system_info_button.connect_clicked({
        let actions_rc = actions_rc.clone();
        move |_| actions_rc.borrow().open_system_info()
    });
    buttons_box.append(&system_info_button);

    let report_bug_button = Button::builder()
    .label("Zgłoś błąd")
    .build();
    report_bug_button.connect_clicked({
        let actions_rc = actions_rc.clone();
        move |_| actions_rc.borrow().report_bug()
    });
    buttons_box.append(&report_bug_button);

    let forum_button = Button::builder()
    .label("Otwórz Forum dyskusyjne")
    .build();
    forum_button.connect_clicked({
        let actions_rc = actions_rc.clone();
        move |_| actions_rc.borrow().open_forum()
    });
    buttons_box.append(&forum_button);

    let update_button = Button::builder()
    .label("Aktualizuj system")
    .build();
    update_button.connect_clicked({
        let actions_rc = actions_rc.clone();
        move |_| actions_rc.borrow().update_system()
    });
    buttons_box.append(&update_button);

    scrolled_window.set_child(Some(&buttons_box));
    main_box.append(&scrolled_window);

    // Separator
    let separator2 = Separator::new(Orientation::Horizontal);
    main_box.append(&separator2);

    // Footer
    let footer_label = Label::builder()
    .label("© 2025 HackerOS Team")
    .css_classes(vec!["footer".to_string()])
    .build();
    main_box.append(&footer_label);

    window.set_content(Some(&main_box));
}
